www/includes/easyparliament/templates/html/section/_calendar.php

```diff
@@ -0,0 +1,166 @@
+  <?php } else { ?>
+    <div class="calendar__header">
+        <?= $monthName ?>
+    </div>
+  <?php } ?>
+
+    <table>
+        <thead>
+            <tr>
+                <th>Mon</th>
+                <th>Tue</th>
+                <th>Wed</th>
+                <th>Thu</th>
+                <th>Fri</th>
+                <th>Sat</th>
+                <th>Sun</th>
+            </tr>
+        </thead>
+        <tbody>
+            <tr>
+
+<?php
+            if ($dayOfWeek > 0) {
+                print "<td colspan=\"$dayOfWeek\">&nbsp;</td>";
+            }
+
+            $currentDay = 1;
+
+            while ($currentDay <= $numberDays) {
+
+                // Seventh column (Sunday) reached. Start a new row.
+
+                if ($dayOfWeek == 7) {
+
+                    $dayOfWeek = 0;
+                    ?></tr>
+                <tr><?php
+                }
+
+                $recess = recess_prettify($currentDay, $month, $year, $recess_major);
+
+                // Is this day actually Today in the real world?
+                // If so, higlight it.
+                // Also highlight days where there are no
+                // sittings - e.g. WH is only Tuesday-Thursday
+                if ($currentDay == $toDay) {
+                    print '<td class="on"';
+                    if ($recess[0] && $recess[0]!=1) print ' title="'.$recess[0].'"';
+                    print '>';
+                } elseif ($recess[0]) {
+                    print '<td class="no"';
+                    if ($recess[0]!=1) print ' title="'.$recess[0].'"';
+                    print '>';
+                } else {
+                    print '<td>';
+                }
+
+                // Is the $currentDay a member of $dates? If so,
+                // the day should be linked.
+                if (in_array($currentDay,$dates)) {
+
+                    $date = sprintf("%04d-%02d-%02d", $year, $month, $currentDay);
+
+                    if ($currentDay == $toDay) {
+                        print '<span>' . $currentDay . '</span></td>';
+                    } else {
+                        $urls['day']->insert(array('d'=>$date));
+                        print "<a href=\"" . $urls['day']->generate() . "\">$currentDay</a></td>";
+                    }
+
+                    // $currentDay is not a member of $dates.
+
+                } else {
+
+                        print '<span>' . $currentDay . '</span></td>';
+                }
+                // Increment counters
+
+                $currentDay++;
+                $dayOfWeek++;
+            }
+
+            // Complete the row of the last week in month, if necessary
+
+            if ($dayOfWeek != 7) {
+
+                $remainingDays = 7 - $dayOfWeek;
+                print "<td colspan=\"$remainingDays\">&nbsp;</td>";
+            }
+?>
+            </tr>
+        </tbody>
+    </table>
+  <?php if(!isset($years)){ ?>
+    <div class="calendar__footer">
+    <?php
+        $y = new URL($this_page);
```

Comment:
This goes wrong on /written-answers-and-statements, and on /debates, not linking to the right URL in most cases.